ocaml/xapi/xapi_pool.ml

```diff
@@ -1620,6 +1620,22 @@ let eject ~__context ~host =
       (fun () -> Xapi_fuse.light_fuse_and_reboot_after_eject ()) ;
     Xapi_hooks.pool_eject_hook ~__context
 
+(** eject [host] from the pool. This code is run on all hosts in the
+pool but only [host] will eject itself. *)
+let eject ~__context ~host =
+  let self_eject = host = Helpers.get_localhost ~__context in
+  let self_master = Pool_role.is_master () in
+  match (self_eject, self_master) with
+  | true, false ->
+      eject ~__context ~host
+  | false, false ->
+      Certificates_sync.eject_certs_from_fs_for ~__context host
+  | false, true ->
+      Certificates_sync.eject_certs_from_fs_for ~__context host ;
+      Certificates_sync.eject_certs_from_db_for ~__context host
```

Comment:
I think it's safer if this line is in message_forwarding.ml - removing the cert from the DB must be the absolute last step, because otherwise calls to `Certificates_sync.eject_certs_from_fs_for ~__context host` will fail

Reply:
That is a good point - will move that back to message forwarding